### DISCUSSION TITLE
OWL use in orchards for Almonds

### DISCUSSION THREAD

**Discussion by Almondfarmer** [geezacoleman/OpenWeedLocator#77] [General]
Good Day, 

As per the title, I want to build an OWL system for our orchard weed sprayer. I have started ordering all the components we need to build the OWL unit, but lead times in South Africa is longer than what I've hoped for. I am building a test unit (only doing 1 unit, if this is successful, I will build the rest of the system including a remote unit inside the cab. 

![Front view](https://user-images.githubusercontent.com/132674040/236500034-46718d18-3064-47fd-bf37-fb0d1df85052.jpeg)
Front view of the boom section. Our booms are around 1.5m wide, thus 2 units will cover each boom. The booms are around 800mm from the ground, thus some modification is needed

![inside view](https://user-images.githubusercontent.com/132674040/236500594-392d732f-8269-43be-b856-772eae34c98f.jpeg)
The current teejet nozzles will be replaced with 12v solenoid Teejet units. I am still looking at options here. 

The sprayer has a Raven 4600 unit in it that has speed and rate settings. This will eventually only be used as a spedo and to ensure that the pressure stays at a set PSI value. (I think) Still trying to figure some stuff out but will post in this thread as we progress. Would adding a LED light next to the OWL unit help for night time spraying?

**Comment by geezacoleman** [comment]
This looks like a really neat use case for the OWL - thanks for sharing!

We usually mount the OWL at 950 - 1000mm so lowering it to 800mm shouldn't be too much of any issue. With the lens we use on the HQ Cam, the field of view is roughly 1:1 - so 800mm high would give you the 800mm wide coverage as you've already mentioned. It looks like you have a good frame to mount the unit off in front of the boom, inside might be challenging. How fast would you be driving? Check out the [newer design/model](https://github.com/geezacoleman/OpenWeedLocator/discussions/19#discussioncomment-3469746) too if you haven't seen it yet, which is a lot more rugged.

![20220904_212657 (2)](https://github.com/geezacoleman/OpenWeedLocator/assets/51358498/41dbf362-3f91-4119-920f-03966269266f)

How many nozzles will it be controlling? Decreasing this is fairly easily done by changing the nozzleNum value as described in #68.  

Night time operation is quite doable with an LED flood light, but may require some adjustment of colour thresholds depending on the quality of light used. We used a [STEDI 90W light](https://www.stedi.com.au/90w-led-flood-light.html) (you can see it in the photo above) and were able to get quite good results at night. I would add one per camera depending on brightness and how far it throws. One challenge is the centre is much more well lit than the edges, so having overlapping lights is advantageous. A benefit of the light in the day is that you balance out the dark shadows from the sun, so it can improve day time performance as well.

Looking forward to seeing how this progresses! Any issues let me know and we can try working them out. Building in-cab control is high on the priority list, so if you have any ideas on a user-interface and what would be useful it would be good to hear them.

**Reply by Minorplanets76** [reply]
How high do you think you can mount the camera? I.e. for standard boom with 0.5m nozzle spacing, do you think you could go up to 2m high and control 4 nozzles?  I assume resolution goes down proportionally, but what about blur and the like?  1:1 ratio would explain why Weedseekers have a camera at each nozzle with a nominal 0.5m height.

**Reply by geezacoleman** [reply]
As you mention resolution becomes (increasingly) the limiting factor if you raise the camera - but we haven't tried lifting it to 2m before. 

I don't see too many issues with large, broadleaf weeds (over 10cm diameter) but performance will suffer on the smallest ones and on grass weeds. The benefit of a higher camera is the motion blur should be reduced given the apparent ground speed will be reduced, but depending on the mounting method some wobble/vibration may also be introduced. Would be neat to try it at least!
